sender/http.go

```diff
@@ -46,13 +48,92 @@ type HTTP struct {
 	Timeout          skogul.Duration `doc:"HTTP timeout."`
 	Insecure         bool            `doc:"Disable TLS certificate validation."`
 	ConnsPerHost     int             `doc:"Max concurrent connections per host. Should reflect ulimit -n. Defaults to unlimited."`
-	IdleConnsPerHost int             `doc:"Mas idle connections retained per host. Should reflect expected concurrency. Defaults to 2 + runtime.NumCPU."`
+	IdleConnsPerHost int             `doc:"Max idle connections retained per host. Should reflect expected concurrency. Defaults to 2 + runtime.NumCPU."`
+	RootCA           string          `doc:"Path to an alternate root CA used to verify server certificates. Leave blank to use system defaults."`
+	ok               bool            // set to OK if init worked. FIXME: Should Verify() check if this is OK? I'm thinking yes.
 	once             sync.Once
 	client           *http.Client
 }
 
+// getCertPool reads the file specified in f and returns a CertPool with
+// the parsed result, suitable for use as RootCAs. If f is empty, it
+// returns nil (with no error), which generally means "use system-wide
+// defaults".
+//
+// Called both during Verify(), and again on the first request. This is
+// done to satisfy Verify()'s requirement of not modifying state and it
+// being optional. We also need Verify to actually test this, so the user
+// can be reasonably certain that a valid configuration is used.
+func getCertPool(f string) (*x509.CertPool, error) {
+	// this means "use system default"
+	if f == "" {
+		return nil, nil
+	}
+	cp := x509.NewCertPool()
+	fil, err := os.Open(f)
```

Comment:
I'd prefer if `f` was called `path` and `fil` could be `f`. Minor nitpick, though.

Reply:
Makes sense. I'll fix it before I merge.